src/main.rs

```diff
@@ -1,58 +1,118 @@
+use clap::{App, SubCommand};
 use dialoguer::{theme::ColorfulTheme, Select};
+use std::collections::HashMap;
 use std::process;
 use std::process::Command;
 
-fn main() {
-    let cmds = [
-        ("Build Application Services", "cargo build"),
+const COMMANDS: &[(&str, (&str, &str))] = &[
+    ("build", ("Build Application Services", "cargo build")),
+    (
+        "test",
         (
             "Run all tests for a pull request",
             "sh ./automation/all_tests.sh",
         ),
+    ),
+    (
+        "test_rust",
         ("Run all Rust tests", "sh ./automation/all_rust_tests.sh"),
+    ),
+    (
+        "verify_desktop",
         (
             "Verify Desktop Environment",
             "sh ./libs/verify-desktop-environment.sh",
         ),
+    ),
+    (
+        "verify_android",
         (
             "Verify Android Environment",
             "sh ./libs/verify-android-environment.sh",
         ),
+    ),
+    (
+        "verify_ios",
         (
             "Verify iOS Environment",
             "sh ./libs/verify-ios-environment.sh",
         ),
+    ),
+    (
+        "smoke_android",
         (
             "Smoke test Android Components",
             "python3 ./automation/smoke-test-android-components.py",
         ),
+    ),
+    (
+        "smoke_fenix",
         (
             "Smoke test Fenix Components",
             "python3 ./automation/smoke-test-fenix.py",
         ),
+    ),
+    (
+        "smoke_ios",
         (
             "Smoke test Firefox iOS Components",
             "python3 ./automation/smoke-test-firefox-ios.py",
         ),
+    ),
+    (
+        "tag_minor",
         (
             "Tag a new release minor release",
             "python3 ./automation/prepare-release.py minor",
         ),
+    ),
+    (
+        "prepare_release",
         (
             "Tag a new release patch release",
             "python3 ./automation/prepare-release.py patch",
         ),
+    ),
+    (
+        "lint_bash",
         (
             "Lint bash script changes",
             "sh ./automation/lint_bash_scripts.sh",
         ),
+    ),
+    (
+        "cargo_update",
         (
             "Create a 'cargo update' PR",
             "python3 ./automation/cargo-update-pr.py",
         ),
-    ];
+    ),
+];
+
+fn main() {
+    let map: HashMap<&str, (&str, &str)> = COMMANDS.iter().cloned().collect();
+    let matches = App::new("asdev")
+        .about("Helps you navigate the Application Services repository")
+        .author("a-s-dev Team")
+        .version("0.1.4")
```

Comment:
I wonder if we can pull this version number from the toml file or something.

Reply:
Oh absolutely. I literally just used env!("CARGO_PKG_VERSION") in my update-notifier 🤦 I'll update this quickly